plugins/core/hu.bme.mit.gamma.ui/src/hu/bme/mit/gamma/ui/taskhandler/VerificationHandler.java

```diff
@@ -309,6 +314,7 @@ public void execute(Verification verification) throws IOException, InterruptedEx
 		traces.addAll(retrievedTraces);
 		if (serializeTraces) { // After 'traces.add...'
 			serializeTraces();
+			MakefileGenerator.tests.clear(); // Manual reset
```

Comment:
This is not a good pattern. Make sure to include the relevant C-related constructs only in the corresponding methods. Maybe the `MakefileGenerator` should be an instantiable class and then, these clears would not need to be carried out.